portfolio/src/main/java/com/google/sps/data/BasketballCourtsDatastore.java

```diff
@@ -0,0 +1,101 @@
+package com.google.sps.data;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonElement;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import javax.servlet.ServletContext;
+import java.util.Scanner;
+
+/** Parses the basketball courts file and returns a list of basketball courts */
+public final class BasketballCourtsDatastore {
+
+  private static final Logger log = Logger.getLogger(BasketballCourtsDatastore.class.getName());
+  public static final String ACCESSIBLE_PROPERTY = "Accessible";
+  public static final String IS_ACCESSIBLE = "Y";
+  public static final String ID_PROPERTY = "Prop_ID";
+  public static final String NAME_PROPERTY = "Name";
+  public static final String LOCATION_PROPERTY = "Location";
+  public static final String NUMCOURTS_PROPERTY = "Num_of_Courts";
+  public static final String LAT_PROPERTY = "lat";
+  public static final String LON_PROPERTY = "lon";
+
+  private BasketballCourtsDatastore() {}
+
+  /** Returns a list of the accessible basketball courts. If there is a bad entry, it should be
+    ignored and the other entries should be returned. */
+  public static List<JsonObject> getAccessibleBasketballCourts(JsonArray allCourts) {
+
+    ImmutableList.Builder<JsonObject> accessibleCourts =
+      new ImmutableList.Builder<JsonObject>();
+
+    for (JsonElement court : allCourts) {
+        JsonObject courtObj = court.getAsJsonObject();
+        JsonElement accessibleEl = courtObj.get(ACCESSIBLE_PROPERTY);
+        if (!accessibleEl.isJsonNull()) {
+          String accessibleStr = accessibleEl.getAsString();
+          if (accessibleStr.equals(IS_ACCESSIBLE)) {
+            try {
+              accessibleCourts.add(createCourtFromJson(courtObj));
+            } catch (IllegalArgumentException e) {
+              log.log(Level.SEVERE, "IllegalArg caught when creating court object from json", e);
+            }
+          }
+        }
+    }
+
+    return accessibleCourts.build();
+  }
+
+  /** Creates a basketball court object given a json object of that court.
+    * Throws {@link IllegalArgumentException} if the fields (not including numOfCourts) are null
+    * or if numOfCourts is not greater than zero. NumOfCourts is null for most entries likely because
+    * that data wasn't reported, but these entries still need to be included so numOfCourts=null is valid. */
+  private static JsonObject createCourtFromJson(JsonObject jsonCourt) {
```

Comment:
well this imethod is not very useful you say "createCourtFromJson" but what you do is take a json object with properties A, B, C and return another json object with properties X, Y, Z. Not sure how useful it is :)